### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by ChenYanting**

### CODE REVIEW

File: src/web/app/contexts/AlertProgressContext.tsx [hunk 1]
```diff
@@ -175,115 +166,46 @@ export class AlertProgressContextProvider extends React.Component<unknown, State
...
       onCancel,
     } = args;
     switch (buttonType) {
       case AlertConstants.YES_NO:
-        onYes = onYes || (() => { });
         onNo = onNo || (() => { });
         buttonLabels = [LANG.yes, LANG.no];
         callbacks = [onYes, onNo];
         primaryButtonIndex = primaryButtonIndex || 0;
-        if (checkbox) {
```
Comment (ChenYanting) [inline src/web/app/contexts/AlertProgressContext.tsx:196]: `checkbox` is never given

File: src/web/app/contexts/AlertProgressContext.tsx [hunk 2]
```diff
@@ -175,115 +166,46 @@ export class AlertProgressContextProvider extends React.Component<unknown, State
...
-          callbacks = [onYes, onNo, ...(callbacks as (() => void)[])];
-        }
         break;
       case AlertConstants.CONFIRM_CANCEL:
-        onConfirm = onConfirm || (() => { });
         onCancel = onCancel || (() => { });
         buttonLabels = [LANG.confirm, LANG.cancel];
         primaryButtonIndex = primaryButtonIndex || 0;
         callbacks = [onConfirm, onCancel];
-        if (checkbox) {
```
Comment (ChenYanting) [inline src/web/app/contexts/AlertProgressContext.tsx:229]: `checkbox` is never given

File: src/web/app/contexts/AlertProgressContext.tsx [hunk 3]
```diff
@@ -175,115 +166,46 @@ export class AlertProgressContextProvider extends React.Component<unknown, State
...
-          checkbox.callbacks = [onCheckedConfirm, onCheckedCancel];
-        }
         break;
       case AlertConstants.RETRY_CANCEL:
-        onRetry = onRetry || (() => { });
         onCancel = onCancel || (() => { });
         buttonLabels = [LANG.retry, LANG.cancel];
         primaryButtonIndex = primaryButtonIndex || 0;
         callbacks = [onRetry, onCancel];
-        if (checkbox) {
```
Comment (ChenYanting) [inline src/web/app/contexts/AlertProgressContext.tsx:241]: `checkbox` is never given

File: src/web/app/contexts/AlertProgressContext.tsx [hunk 4]
```diff
@@ -175,115 +166,46 @@ export class AlertProgressContextProvider extends React.Component<unknown, State
...
-          const onCheckedRetry = checkbox.onRetry ? checkbox.onRetry : () => { };
-          const onCheckedCancel = checkbox.onCancel ? checkbox.onCancel : () => { };
-          checkbox.callbacks = [onCheckedRetry, onCheckedCancel];
-        }
         break;
       case AlertConstants.CUSTOM_CANCEL:
         onCancel = onCancel || (() => { });
         primaryButtonIndex = primaryButtonIndex || 0;
-        if (!buttonLabels) {
```
Comment (ChenYanting) [inline src/web/app/contexts/AlertProgressContext.tsx:250]: `buttonLabels` is always given in the case of `CUSTOM_CANCEL` button type

**Review by dkdkwizard — APPROVED** [review]

**Pull request merged**